Snowflake.Data/Core/Session/SessionPool.cs

```diff
@@ -536,9 +552,20 @@ public int GetCurrentPoolSize()
             return _idleSessions.Count + _busySessionsCounter.Count() + _sessionCreationTokenCounter.Count();
         }
 
+        public SessionPoolState GetCurrentState()
+        {
+            return new SessionPoolState(
+                _idleSessions.Count,
+                _busySessionsCounter.Count(),
+                _sessionCreationTokenCounter.Count(),
+                _waitingForIdleSessionQueue.WaitingCount(),
+                IsMultiplePoolsVersion()
+            );
+        }
+
         public bool SetPooling(bool isEnable)
         {
-            s_logger.Info($"SessionPool::SetPooling({isEnable})");
+            s_logger.Info($"SessionPool::SetPooling({isEnable})" + PoolIdentification());
```

Comment:
there are other public interfaces which do not have this kind of info

Reply:
which one? where would you like to add logging?

Reply:
I added only to setters